src/app/components/book-dialog/add/add-dialog.component.ts

```diff
@@ -0,0 +1,55 @@
+import { Component, Inject, ChangeDetectionStrategy, EventEmitter } from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { IReadBook } from '@app/shared/models/read-book.model';
+import { FormActions } from '@app/shared/enums';
+
+@Component({
+  selector: 'app-add-dialog',
+  templateUrl: './add-dialog.component.html',
+  styleUrls: ['../book-dialog.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class AddDialogComponent {
+  private readonly _action = FormActions.Add;
+  private _newBook: IReadBook;
+  private _formInvalid: boolean;
+  private _dialogClosed: EventEmitter<boolean> = new EventEmitter();
```

Comment:
Why do we need this emitter?

Reply:
To communicate between AddDialogComponent and inner BookFormComponent. I need to pass form value to the base component (BookListComponent) before these 2 components get destroyed.